### PR TITLE
Finished off Item's mappings. Also mapped some itemblock stuff. 

### PR DESCRIPTION
**Pull request by Darkhax**
This PR finishes off all mappings in Item

### CODE REVIEW

File: mappings/net/minecraft/item/Item.mapping [hunk 1]
```diff
@@ -10,24 +39,81 @@ CLASS none/aeo net/minecraft/item/Item
...
+		METHOD a apply (Lnone/aeq;)Ljava/lang/String;
+			ARG 0 stack
+	CLASS none/aeo$20
+		METHOD a apply (Lnone/aeq;)Ljava/lang/String;
+			ARG 0 stack
+	CLASS none/aeo$21
+		METHOD a apply (Lnone/aeq;)Ljava/lang/String;
+			ARG 0 stack
 	FIELD a BLOCK_ITEM_MAP Ljava/util/Map;
+	FIELD b GETTER_DAMAGED Lnone/aer;
```
Comment (shadowfacts) [inline mappings/net/minecraft/item/Item.mapping:97]: `GETTER_X` or `X_GETTER`? I personally prefer `X_GETTER`

File: mappings/net/minecraft/item/Item.mapping [hunk 2]
```diff
@@ -98,24 +186,32 @@ CLASS none/aeo net/minecraft/item/Item
...
 		ARG 0 block
 		ARG 1 itemBlock
 	METHOD a canHarvestBlock (Lnone/asj;)Z
 		ARG 0 state
-	METHOD a (Lnone/dt;)Z
+	METHOD a readTag (Lnone/dt;)Z
+		ARG 0 tag
+	METHOD a getProperty (Lnone/kp;)Lnone/aer;
+		ARG 0 id
+	METHOD a addProperty (Lnone/kp;Lnone/aer;)V
```
Comment (shadowfacts) [inline mappings/net/minecraft/item/Item.mapping:204]: Shouldn't this be `setProperty`?

**Comment by shadowfacts** [comment]
Other than that, LGTM

File: mappings/net/minecraft/item/Item.mapping (hunk 1 above)
Comment (darkevilmac) [reply mappings/net/minecraft/item/Item.mapping:97]: Prefixing with getter makes things a bit easier as far as usage in an ide goes.

File: mappings/net/minecraft/item/Item.mapping (hunk 2 above)
Comment (Darkhax) [reply mappings/net/minecraft/item/Item.mapping:204]: I went with addProperty, because the method adds it to a list rather than setting a field.

Comment (shadowfacts) [reply mappings/net/minecraft/item/Item.mapping:204]: Aha, that's good then

**Comment by asiekierka** [comment]
LGTM

File: mappings/net/minecraft/item/Item.mapping (hunk 1 above)
Comment (Darkhax) [reply mappings/net/minecraft/item/Item.mapping:97]: I went with `GETTER_X` for the reason mentioned above. That has been our naming scheme for a lot of things, like `MODIFIER_X` and `ItemX`

**Pull request merged**